pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -283,3 +292,21 @@ func findNewName(args []string, oldRc *api.ReplicationController) string {
 	}
 	return ""
 }
+
+func isReplicasDefaulted(info *resource.Info) bool {
+	if info == nil || info.VersionedObject == nil {
+		// was unable to recover versioned info
+		return false
+	}
+	switch info.Mapping.APIVersion {
+	case "v1":
+		if rc, ok := info.VersionedObject.(*v1.ReplicationController); ok {
```

Comment:
The reason we need a versioned object here is because at this point info.Object has already been defaulted. If that field wasn't provided, it will either have 0 replicas per v1beta3 defaults or 1 replica per v1 defaults.